src/Core/Authentication.php

```diff
@@ -6,69 +6,378 @@
 
 namespace Friendica\Core;
 
+use Exception;
 use Friendica\App;
-use Friendica\BaseObject;
-use Friendica\Network\HTTPException\ForbiddenException;
+use Friendica\Core\Config\Configuration;
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Model\User;
+use Friendica\Network\HTTPException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Network;
+use Friendica\Util\Strings;
+use LightOpenID;
+use Friendica\Core\L10n\L10n;
+use Psr\Log\LoggerInterface;
 
 /**
  * Handle Authentification, Session and Cookies
  */
-class Authentication extends BaseObject
+class Authentication
```

Comment:
Almost all of the method of this new class take and `App` as first argument, which makes me think these methods should be moved to `App`.